processor/spanmetricsprocessor/processor.go

```diff
@@ -225,9 +229,17 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces ptrace.Traces)
 func (p *processorImp) tracesToMetrics(ctx context.Context, traces ptrace.Traces) error {
 	p.lock.Lock()
 
+	// Make sure metricKeys are empty in current batch
+	if len(p.metricKeys) != 0 {
+		p.metricKeys = []metricKey{}
+	}
+
 	p.aggregateMetrics(traces)
 	m, err := p.buildMetrics()
 
+	// Clean current batch metricKeys
+	p.metricKeys = []metricKey{}
+
```

Comment:
Since `p.metricKeys` are reset above, I don't think we need to clean up the batch since it's no used from now on, and I don't think it would have such a large memory footprint. 

What do you think?

Reply:
I would ask a more deeper question: Why do you need this as a member in the struct?

Reply:
it can simplity the pass of param, if use a method scope param, I need pass it throught 3 layer methods to use it.